scripts/lint_prowjobs.go

```diff
@@ -0,0 +1,279 @@
+	Bucket                   string
+	Cluster                  string
+	ServiceAccountName       string
+	DefaultMakeTarget        string
+	CniMakeTarget            string
+	HelmMakeTarget           string
+	ReleaseToolingMakeTarget string
+}
+
+type PostsubmitConstants struct {
+	Bucket                       string
+	Cluster                      string
+	DefaultServiceAccountName    string
+	AthensHelmServiceAccountName string
+	DocsServiceAccountName       string
+	MakeTarget                   string
+}
+
+type Presubmits struct {
+	PresubmitJobs map[string][]prowapi.Presubmit `json:"presubmits,omitempty"`
+}
+
+type Postsubmits struct {
+	PostsubmitJobs map[string][]prowapi.Postsubmit `json:"postsubmits,omitempty"`
+}
+
+var presubmitErrors map[string][]string
+var postsubmitErrors map[string][]string
+var presubmitErrorCount, postsubmitErrorCount = 0, 0
+
+func main() {
+	var presubmitType Presubmits
+	var postsubmitType Postsubmits
+	presubmitFiles := []string{}
+	postsubmitFiles := []string{}
+	presubmitConstants := PresubmitConstants{
+		Bucket:                   "s3://prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp",
+		Cluster:                  "prow-presubmits-cluster",
+		ServiceAccountName:       "presubmits-build-account",
+		DefaultMakeTarget:        "build",
+		CniMakeTarget:            "release",
+		HelmMakeTarget:           "verify",
+		ReleaseToolingMakeTarget: "test/build",
+	}
+	postsubmitConstants := PostsubmitConstants{
+		Bucket:                       "s3://prowdataclusterstack-316434458-prowbucket7c73355c-1n9f9v93wpjcm",
+		Cluster:                      "prow-postsubmits-cluster",
+		DefaultServiceAccountName:    "postsubmits-build-account",
+		AthensHelmServiceAccountName: "charts-build-account",
+		DocsServiceAccountName:       "docs-build-account",
+		MakeTarget:                   "release",
+	}
+
+	pullBaseSha := os.Getenv("PULL_BASE_SHA")
+	pullPullSha := os.Getenv("PULL_PULL_SHA")
+	cmdOut, err := exec.Command(`git`, "diff", "--name-only", pullBaseSha, pullPullSha).Output()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "There was an error running the git command: ", err)
+		os.Exit(1)
+	}
+	output := string(cmdOut)
+	diffFiles := strings.Fields(output)
+	for _, file := range diffFiles {
+		if strings.Contains(file, "presubmits") {
+			presubmitFiles = append(presubmitFiles, file)
+		}
+		if strings.Contains(file, "postsubmits") {
+			postsubmitFiles = append(postsubmitFiles, file)
+		}
+	}
+
+	for _, presubmitFile := range presubmitFiles {
+		presubmitChecks(presubmitFile, presubmitConstants, &presubmitType)
+	}
+
+	for _, postsubmitFile := range postsubmitFiles {
+		postsubmitChecks(postsubmitFile, postsubmitConstants, &postsubmitType)
+	}
+
+	fmt.Println()
+	for presubmitFile, errors := range presubmitErrors {
+		if len(errors) > 0 {
+			fmt.Println(presubmitFile)
+			for _, errorString := range errors {
+				presubmitErrorCount++
+				fmt.Printf("  %s\n", errorString)
+			}
+		}
+		fmt.Println()
+	}
```

Comment:
I'd suggest making this a separate printer function (or method on a type) where you pass the files/errors. You could invoke it once for each type of check (pre/post submit)

Reply:
For the printer, I'd suggest looking at the [`tabwriter`](https://golang.org/pkg/text/tabwriter/) package